src/units/unit.hpp

```diff
@@ -1898,6 +1898,33 @@ class unit : public std::enable_shared_from_this<unit>
 	 */
 	bool ability_matches_filter(const config & cfg, const std::string& tag_name, const config & filter) const;
 
+	/**
+	 * Return true, if unit calling function and unit designated by @ self_underlying_id are different units including when either is a fake_unit generated from the other one, if unit not incapacitated and own a ability with [affect_adjacent] subtag
```

Comment:
This doesn't feel right to me, it's putting together three different concepts.

The id check is commutative - if A has a different id to B then B has a different id to A. The "when either ... from the other one" text suggests that it doesn't matter which unit is A and which is B, which is true for the id part, but false for the incapacitated() and has_ability_distant() checks.

Having a function that combines `!incapacitated() && has_ability_distant()` doesn't feel right to me either - when it's checking `incapacitated()` it's handling one way in which the ability might be inactive, but not others. It makes sense that the three checks would be on the same line in the caller, but I feel they aren't similar enough to put them into a separate function.